clients/drcachesim/scheduler/scheduler.cpp

```diff
@@ -2335,6 +2340,15 @@ scheduler_tmpl_t<RecordType, ReaderType>::next_record(output_ordinal_t output,
                 lock.lock();
                 VPRINT(this, 5, "next_record_mid[%d]: switching from %d to %d\n", output,
                        prev_input, outputs_[output].cur_input);
+                if (!preempt) {
+                    if (options_.quantum_unit == QUANTUM_INSTRUCTIONS &&
+                        record_type_is_instr(record)) {
+                        --inputs_[prev_input].instrs_in_quantum;
+                    } else if (options_.quantum_unit == QUANTUM_TIME) {
+                        inputs_[prev_input].time_in_quantum -=
```

Comment:
Looks like time_in_quantum is really the time_spent_in_quantum? And prev_time_in_quantum is the snapshot of the last timestamp? Maybe rename s/time_in_quantum/time_spent_in_quantum/ to make the distinction clearer?

Reply:
Done.